dev/ci/user-overlays/README.md

```diff
@@ -65,6 +65,15 @@ named `more_efficient_tc`.
 - From the `<job name>` subdirectory, commit your changes to a new
   branch, based on the base branch name listed in `ci-basic-overlay.sh`,
   for example `coq-master` for elpi.
+  - If the project in `ci-basic-overlay.sh` includes a `submodule_directory`
+    argument (for example, the last argument of `project bedrock2
+    "https://github.com/mit-plv/fiat-crypto" "master" "" "rupicola/bedrock2"`),
+    you will need to navigate to the corresponding subfolder of the base
+    repository to find out what commit the project is based on. The
+    `.gitmodules` folder of the parent repo may contain a `branch =`
+    specification for some of its submodules, indicating the base branch name.
+    For example, [`bedrock2` indicates `branch = rv32i` for
+    `kami`](https://github.com/mit-plv/bedrock2/blob/8cc35c50ec7f302e621d23898b007e999116641f/.gitmodules#L7-L10).
```

Comment:
This sounds complicated, not sure we want to merge.

Reply:
This is just documenting what is already the case, except that today there is no guidance on how to make overlays.  In fact the situation with this PR is significantly simpler: `_build_ci/<job name>` will be a symlink to the appropriate folder, and this complicated procedure is only needed to determine the branch against which the PR should be made.  (This procedure can also be used to determine the repository, though this can also be done with `git remote -v`)

Maybe the text could make this clearer?

Reply:
@SkySkimmer what do you think of this?
```suggestion
  - If the project in `ci-basic-overlay.sh` includes a `submodule_directory`
    argument (for example, the last argument of `project bedrock2
    "https://github.com/mit-plv/fiat-crypto" "master" "" "rupicola/bedrock2"`),
    you can list all branches the current commit is on with
    `git fetch; git branch -r --contains $(git rev-parse HEAD)`
    (use `HEAD^` instead of `HEAD` if you have already committed), and you can
    see if the parent repository contains a branch specification by executing
    `(cd "$(pwd)/.."; git config -f .gitmodules --get-regexp '^submodule\..*\.branch')`
    from the `_build_ci/<job name>` directory.
```

Reply:
TBH I wanted to know what this PR changes in the overlay workflow ("This should hopefully provide a relatively seamless experience for overlay writers" is not very specific) and only saw a bunch of added stuff which was not encouraging.

Reply:
I think finding the correct branch should not be made to sound this complex, something like "for projects which are submodules, to determine the branch to open a PR against you should look in the parent project's repository" is probably enough.

Reply:
Sorry about that @SkySkimmer 

For projects that are not submodules / where the overlay does not make changes to any submodules, this PR changes nothing.

For overlays to submodules of projects, the old overlay workflow included additional steps not present in the documentation:
1. cd to the submodule directory
2. make the commit there
3. push to a branch
4. figure out the appropriate repo and branch to make the PR to
5. make the PR
6. make corresponding commits, branches, and PRs to bump the submodule in the parent project (recursively if necessary)
7. write another version of the submodule bump commit that points the submodule at your fork (do this recursively if necessary)
8. write an overlay file

This PR removes all of these steps, making the workflow for writing overlays to submodules identical to the workflow for writing overlays to other projects, with the exception of two details:
1. the repository to make the PR against and the branch name cannot be read off the `ci-basic-overlay` file but must instead be obtained from `git remote -v` and either `git fetch; git branch -r --contains $(git rev-parse HEAD)` or looking at the parent repository
2. when the PR is merged, upstream must merge not only the overlay PR, but also (recursive) submodule bumping PRs (otherwise Coq master will pick up the old versions).  I think part of the contract of using submodules should be that upstream maintainers will not just merge the overlay PRs promptly (as is required of all developments today), but will take care of bumping required submodules promptly on their end, without any additional input from Coq developers.*

As an alternative to (1), we can demand that submodule projects include a comment with this information alongside the project declaration in `ci-basic-overlay`, in which case the only changes for submodule projects are:
1. Determine the upstream repo and branch from the comment rather than the `project` line
2. Tell upstream to take care of merging not just the overlay but also of bumping submodules

*Note that this is a relatively automated process, and so even coq-community projects using submodules would not incur too much cost; we simply tell dependabot to check for updates (go to, e.g., https://github.com/mit-plv/fiat-crypto/network/updates, click "recent update jobs" next to `.submodules`, click "check for updates") and then say `@dependabot squash and merge` on the PRs it creates.

Reply:
I realized I had not updated `dev/tools/create_overlays.sh`; I added 8 loc to have the generated overlay point to the correct repo and another 8 loc to make it print a comment instructing the developer where to make the PR against.

So now the only differences are:
1. Determine the upstream repo and branch from the end of the project line rather than the middle; or determine them from the comment in the generated overlay file
2. Tell upstream to take care of merging not just the overlay but also of bumping submodules